dss/subscriptions_v2/__init__.py

```diff
@@ -30,10 +48,17 @@ def put_subscription(doc: dict):
 
 def get_subscription(replica: Replica, owner: str, uuid: str):
     try:
-        item = dynamodb.get_item(table=subscription_db_table.format(replica.name),
-                                 hash_key=owner,
-                                 sort_key=uuid)
-        return json.loads(item)
+        item = dynamodb.get_all_key_attributes(table=subscription_db_table.format(replica.name),
+                                               hash_key=owner,
+                                               sort_key=uuid)
+        payload = json.loads(item['body'])
+        stats = {}  # type: typing.Dict
+        for attribute_type in[SubscriptionStats.ATTEMPTS, SubscriptionStats.SUCCESSFUL, SubscriptionStats.FAILED]:
+            attribute_value = item.get(attribute_type, None)
+            if attribute_value:
+                stats[attribute_type] = attribute_value
+        payload[SubscriptionData.STATS] = stats
+        return payload
     except dynamodb.DynamoDBItemNotFound:
         return None
```

Comment:
The try-except block should be scoped only to lines where the exception is expected. In this case, there is seemingly no functional difference if this try-except block only wraps the `dynamodb.get_all_key_attributes` call, but I still think it's worth fixing nonetheless.

Reply:
addressed